dev_guide/pod_preset.adoc

```diff
@@ -0,0 +1,567 @@
+enabling customers to test functionality and provide feedback during the
+development process.
+
+For more information on Red Hat Technology Preview features support scope, see
+https://access.redhat.com/support/offerings/techpreview/.
+endif::[]
+====
+
+Using pod preset objects you can inject:
+
+* xref:../dev_guide/secrets.adoc#dev-guide-secrets[secret objects]
+* xref:../dev_guide/configmaps.adoc#consuming-configmap-in-pods[`ConfigMap` objects]
+* xref:../dev_guide/volumes.adoc#dev-guide-volumes[storage volumes]
+* container volume mounts
+* environment variables
+
+Developers only need make sure the pod labels match the label selector on the PodPreset in order to add all that information to the pod. The xref:../architecture/core_concepts/pods_and_services.adoc#labels[label] on a pod associates the pod with one or more pod preset objects that have a matching xref:../architecture/core_concepts/pods_and_services.adoc#services[label selectors].
+
+Using pod presets, a developer can provision pods without needing to know the details about the services the pod will consume. An administrator can keep configuration items of a service invisible from a developer without preventing the developer from deploying pods. For example, an administrator can create a pod preset that provides the name, user name, and password for a database through a secret and the database port through environment variables. The pod developer only needs to know the label to use to include all the information in pods. A developer can also create pod presets and perform all the same tasks. For example, the developer can create a preset that injects environment variable automatically into multiple pods.
+
+When a pod preset is applied to a pod, {product-title} modifies the pod specification, adding the injectable data and annotating the pod spec to show that it was modified by a pod preset. The annotation is of the form:
+
+----
+podpreset.admission.kubernetes.io/<pod-preset name>: `resource version`
+----
+
+You can configure a pod specification to prevent this modification using the `podpreset.admission.kubernetes.io/exclude: "true"` annotation in the pod specification, as shown in the pod specification example below.
+
+In order to use pod presets in your cluster:
+
+* An administrator must xref:dev-guide-pod-presets-adm[enable the pod preset admission controller plug-in] through the *_/etc/origin/master/master-config.yaml_*;
+* The pod preset author must enable the API type `settings.k8s.io/v1alpha1/podpreset` through the pod preset and add injectable information to the pod preset.
+
+If the pod creation encounters an error, the pod is created without any injected resources from the pod preset.
+
+[NOTE]
+====
+The Pod Preset feature is available only if the Service Catalog has been installed.
+====
+//end::pod-preset[]
+
+.Sample pod preset object
+----
+kind: PodPreset
+apiVersion: settings.k8s.io/v1alpha1 <1>
+metadata:
+  name: allow-database <2>
+  namespace: myns
```

Comment:
remove namespace line, 
for OpenShift user, project/namespace is auto-generated according to the project working on, or specified by -n [$project]
Kubernetes user may prefer namespace in spec.
@derekwaynecarr correct me if I am wrong. thanks.